beta-20170420.md

```diff
@@ -0,0 +1,85 @@
+    <a href="https://binaries.cockroachdb.com/cockroach-beta-20170420.windows-6.2-amd64.exe.tgz"><button id="windows" data-eventcategory="windows-binary-release-notes">Windows</button></a>
+</div>
+
+### Backwards-Incompatible Changes
+
+- Security-related command-line arguments have been changed. Instead of individual flags, certificates and keys are now loaded from `--certs-dir` (defaults to `~/.cockroach-certs`, or settable with `COCKORACH_CERTS_DIR` environment variable). The individual flags still work in this release but will be removed in the next one. The `--insecure` flag is now *required* for insecure mode. [#14703](https://github.com/cockroachdb/cockroach/pull/14703)
+- The `--alsologtostderr` argument has been renamed to `--logtostderr`, and now takes an optional argument to specify a threshold. [#14841](https://github.com/cockroachdb/cockroach/pull/14841)
+- The `^` operator now means `pow()` instead of bitwise xor (this is consistent with PostgreSQL). [#14882](https://github.com/cockroachdb/cockroach/pull/14882)
+
+### Upgrade Notes
+
+- This release cannot run at the same time as releases prior to `beta-20170413`. If you are running an older version, you must either upgrade to that version first or take down all servers in the cluster before starting any server with this version. This release includes new functionality to make future upgrades smoother. [#14959](https://github.com/cockroachdb/cockroach/pull/14959)
+
+### Build Changes
+
+- CockroachDB's build system has been rewritten to improve portability. It can no longer be installed with `go get` alone; use of the `Makefile` is now required. GNU Make, CMake, and XZ Utils must also be installed to build from source. As a result, CockroachDB can now be built on a larger variety of systems, including experimental Windows support. [#14840](https://github.com/cockroachdb/cockroach/pull/14840)
+
+### SQL Language Changes
+
+- Type errors in `CHECK` constraints are now detected earlier. [#14984](https://github.com/cockroachdb/cockroach/pull/14984)
+- Aggregation functions now work correctly above the windowing level. [#14729](https://github.com/cockroachdb/cockroach/pull/14729)
+- The default precision for inexact decimal operations is now 20 digits (up from 16). [#15001](https://github.com/cockroachdb/cockroach/pull/15001)
+- Bounds are now checked when converting `FLOAT` or `DECIMAL` values to `INTEGER`. [#14967](https://github.com/cockroachdb/cockroach/pull/14967)
+- The `pg_catalog` data for `int2vector` is now consistent with PostgreSQL. [#14989](https://github.com/cockroachdb/cockroach/pull/14989)
+
+### Command-Line Interface Changes
+
+- New arguments `--log-file-max-size` (default 10 MiB) and `--log-dir-max-size` (default 100 MiB) can be used to control log rotation and deletion. `--log-file-verbosity` (default `INFO`) can be used to control the verbosity of the log file. [#14841](https://github.com/cockroachdb/cockroach/pull/14841)
```

Comment:
#14703 also reworks all the `cockroach cert` commands. I don't think the details need to be added here, but a mention doesn't hurt.